.github/workflows/ci.yml

```diff
@@ -0,0 +1,17 @@
+name: automatically deploy mkdocs site 
+on:
+  push:
+    branches:
+      - master 
+      - main
+      - mkdocs
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+      - run: pip install mkdocs "mkdocstrings[python]" mkdocs-material 
+      - run: mkdocs gh-deploy --force
```

Comment:
Just in case the PR merge has some issues. In order for this workflow to be implemented, your [personal access token](https://github.com/settings/tokens) needs to authenticate workflows as well.

Reply:
I missed this comment! What should I do here?

Reply:
I'm incorporating a github workflow into this pull request so the mkdocs site gets built automatically whenever there's a change in the repository. However, that means your personal access token needs to be able to authenticate workflows. I wrote the original message to let you know about this just in case the merge encounters an error.

Reply:
Created the access token for GitHub workflows, do I need to set it somewhere?

Reply:
I just modified my existing access token to allow for GitHub workflows. I suggest you try to merge this PR first and see if the workflow here is allowed.